### ISSUE TITLE
Update Uplink-c depenency to v1.6.2

### ISSUE THREAD

**Issue by ifraixedes** [storj-thirdparty/uplink-rust#12]
A new version of uplink-c has been released, the [v1.6.2](https://github.com/storj/uplink-c/releases/tag/v1.6.2)

- [ ] Update the `uplink-sys` crate to use this new uplink-c dependency. The new version is `v0.3.1`.
- [ ] Update the `uplink` crate to use the new uplink-sys version. Do this in the branch [uplink-first-version](https://github.com/storj-thirdparty/uplink-rust/tree/uplink-first-version).
- [ ] Publish the new `uplink-sys` crate in crates.io.

Blocked by #9

**Comment by ifraixedes** [comment]
Superseded by #15
